lib/catalog/approval_transition.rb

```diff
@@ -39,28 +37,29 @@ def state_transitions
 
     def submit_order
       finalize_order
-      @order_item.update_message("info", "Submitting Order #{@order_item.order_id} for provisioning ")
+      @order_item.order.update_message("info", "Submitting Order for provisioning")
       Catalog::SubmitNextOrderItem.new(@order_item.order_id).process
     rescue ::Catalog::TopologyError => e
       Rails.logger.error("Error Submitting Order #{@order_item.order_id}, #{e.message}")
-      @order_item.update_message("error", "Error Submitting Order #{@order_item.order_id}, #{e.message}")
+      @order_item.order.update_message("error", "Error Submitting Order: #{e.message}")
```

Comment:
I think we do not need this message here because the order will still continue with next order item. The order will eventually fail after all items finish.

On the other hand I found nowhere that we added a progress message to order indicating the order has finishes. Maybe we can add it before https://github.com/RedHatInsights/catalog-api/blob/master/lib/catalog/order_state_transition.rb#L11.

Reply:
I have added the order completed message at: https://github.com/RedHatInsights/catalog-api/pull/866/files#diff-5713803047da32d147c111172c4e3353f312b6768e84dfb7a04a32d3f27754b5R25

Reply:
We want to add message for other abnormal finished orders, i.e., canceled and failed. But make sure we do not mark an order finished too early, i.e., do this until all items finished.

Reply:
Even order still continues, I feel it's still helpful to record this message in order detail since the messages in provision tab are not accessible to all users.